packages/slice-machine/next.config.mjs

```diff
@@ -1,30 +1,35 @@
-import * as fs from "node:fs";
-import * as path from "node:path";
-import * as url from "node:url";
 import { withSentryConfig } from "@sentry/nextjs";
 import { createVanillaExtractPlugin } from "@vanilla-extract/next-plugin";
 import semver from "semver";
 
-const __dirname = path.dirname(url.fileURLToPath(import.meta.url));
+import pkg from "./package.json" assert { type: "json" };
 
-/** @type {{ version: string }} */
-const pkg = JSON.parse(
-  fs.readFileSync(path.resolve(__dirname, "package.json"), "utf8"),
-);
-
-/** @type string */
+const parsedPkgVersion = semver.parse(pkg.version);
+if (parsedPkgVersion === null) {
+  throw new Error(
+    `Package \`${pkg.name}\` has an invalid version \`${pkg.version}\` in its manifest.`,
+  );
+}
 
-const RELEASE_NUMBER = pkg.version;
-const isStableVersion =
-  /^\d+\.\d+\.\d+$/.test(RELEASE_NUMBER) && semver.lte("0.1.0", RELEASE_NUMBER);
+let sentryEnvironment;
+if (parsedPkgVersion.prerelease.length === 0) {
+  sentryEnvironment = process.env.NODE_ENV;
+} else if (
+  parsedPkgVersion.prerelease[0] === "alpha" ||
+  parsedPkgVersion.prerelease[0] === "beta"
+) {
+  sentryEnvironment = parsedPkgVersion.prerelease[0];
+} else {
+  throw new Error(
+    `Invalid package version: \`${pkg.name}@${parsedPkgVersion.version}\`. The first prerelease component \`${parsedPkgVersion.prerelease[0]}\` must be either \`alpha\` or \`beta\`.`,
+  );
+}
```

Comment:
Since this code is duplicated in `@slicemachine/init`, `slice-machine-ui`, and `start-slicemachine`, should we abstract it to a common location?

Maybe it could be in `@slicemachine/manager` as a dedicated export:

```ts
import { getStaticEnvironmentInfo } from "@slicemachine/manager";
import * as pkg from "@slicemachine/manager";

getStaticEnvironmentInfo(pkg);
//  => { version: string; environment: string }
```